src/main/java/com/vmware/vipclient/i18n/base/cache/Cache.java

```diff
@@ -110,5 +119,62 @@
      * @return a drop id
      */
     public String getDropId();
+    
+    public class CacheItem {
+    	public CacheItem() {
+    		
+    	}
+    	
+    	public CacheItem (Map<String, String> dataMap, final Map<String, Object> cacheProps) {
+    		super();
+    		this.addCachedData(dataMap);
+    		this.addCacheProperties(cacheProps);
+    	}
+    	
+    	public CacheItem (Map<String, String> dataMap) {
+    		super();
+    		this.addCachedData(dataMap);
+    	}
+    	
+    	private final Map<String, String> cachedData = new HashMap<String, String>();
+
+		/*
+    	 * A map of properties associated to the cachedData (e.g. etag and cache control)
+    	 * This map is optional and will not be instantiated if not needed.
+    	 */
+    	private Map<String, Object> cacheProperties;
+        
+        public Map<String, String> getCachedData() {
+			return cachedData;
+		}
+        
+        public void addCachedData(Map<String, String> cachedData) {
+			if (cachedData != null) {
```

Comment:
It seems I'm wrong, this.cachedData can't be null, but parameter 'cachedData' may be null. The judge is needed still.

Reply:
This is because the existing code returns null value if http code fails, and actually saves the null value in cache. I agree with you, this is wrong! But I do not want to change the return of the old code because it is our of scope of this PR. But in this PR, I do not want to store null values in the cache so for now, put an empty map there that could be populated later.

Reply:
Any suggestions?

Reply:
You can see this in the usage of putAll
     * @throws NullPointerException if the specified map is null
So don't pass null value to the Map.putAll method.